### PR TITLE
fix: Allow JPEG and GIF images as profile photos #2332

### PR DESCRIPTION
**Pull request by islathehut**
* Fixes `UserProfileStore` and `UserProfileContextMenu` to allow JPEG and GIF profile photos as well as PNG.
* Updates E2E tests to include JPEG and GIF cases on profile photo tests
* Adds new backend unit tests for img validation
* Adds new `npm` scripts at the root for conveniently running the desktop app locally instead of having to `cd` into the desktop package folder and linting all packages at once.
* Adds new `npm` scripts for local binary compilation and updates E2E tests allow using local binaries
* Adds documentation for new local binary compilation and testing scripts
* Fixes issues I had while running tor tests locally
* Fix the issue with `ContextMenuProps` having type `unknown` for the field `visible`

GitHub Issue: https://github.com/TryQuiet/quiet/issues/2332

### Pull Request Checklist

- [x] I have linked this PR to a related GitHub issue.
- [x] I have added a description of the change (and Github issue number, if any) to the root [CHANGELOG.md](https://github.com/TryQuiet/quiet/blob/develop/CHANGELOG.md).

### (Optional) Mobile checklist

Please ensure you completed the following checks if you did any changes to the mobile package:

- [ ] I have run e2e tests for mobile
- [ ] I have updated base screenshots for visual regression tests

### CODE REVIEW

**Comment by islathehut** [comment]
Fixes #2332

**Review by leblowl — APPROVED** [review]
Thanks. I would move the changelog changes under unreleased, but I can also do it later. Here are some of my thoughts around validation for this feature: https://github.com/TryQuiet/quiet/discussions/2173 in case you are interested ... perhaps the image magic byte sequence check is unnecessary, but it doesn't seem like a big deal.

**Comment by islathehut** [comment]
> Thanks. I would move the changelog changes under unreleased, but I can also do it later. Here are some of my thoughts around validation for this feature: #2173 in case you are interested ... perhaps the image magic byte sequence check is unnecessary, but it doesn't seem like a big deal.

Updated the CHANGELOG files with that formatting.  I think for now we can leave the magic byte checks but would be curious if we could do away with it while still maintaining assurance that images are what they say they are.  Part of me also wonders if we should merge the type check (e.g. `data:image/png;base64`) with the magic byte check for performance (since we'll only have to run one header check per image) and some extra assurance (that way we are validating that not only is the type correct on the uploaded image but it matches the magic byte).

**Comment by islathehut** [comment]
Refactored the photo validation so its consolidated and connected the file type check with the magic number check for added assurance/performance.

**Comment by EmiM** [comment]
Could you add backend unit tests for new file types?

File: packages/e2e-tests/src/tests/userProfile.test.ts [hunk 1]
```diff
@@ -106,11 +106,25 @@ describe('User Profile Feature', () => {
...
+  })
+
+  it('Owner updates their profile photo with JPEG', async () => {
+    const menu = new UserProfileContextMenu(users.owner.app.driver)
+    await menu.openMenu()
+    await menu.openEditProfileMenu()
+    await menu.uploadJPEGPhoto()
+  })
+
+  it('Owner updates their profile photo with GIF', async () => {
```
Comment (EmiM) [inline packages/e2e-tests/src/tests/userProfile.test.ts:120]: Could you check locally if userProfile test is passing?
* Last of "Owner updates their profile photo with..." tests uploads gif photo so if the upload suppose to be successful then assertion in "Owner's message with profile photo is visible on channel" should be updated from checking png to gif
* Could you maybe add assertion at the end of each of "Owner updates their profile photo with..." cases to make sure that e.g users do not see upload error message?

**Review by EmiM — CHANGES_REQUESTED** [review]

File: packages/e2e-tests/src/tests/userProfile.test.ts (hunk 1 above)
Comment (islathehut) [reply packages/e2e-tests/src/tests/userProfile.test.ts:120]: Added assertions and testing locally now

**Review by EmiM — APPROVED** [review]

**Pull request merged**